cmd/statusd/main.go

```diff
@@ -158,6 +166,34 @@ func startCollectingStats(interruptCh <-chan struct{}, nodeManager common.NodeMa
 	}
 }
 
+func syncAndStopNode(nodeManager common.NodeManager, timeout int) (exitCode int) {
+	log.Println("Node will synchronize and exit")
+	if timeout < 0 {
+		log.Println("Sync and stop error: negative timeout value")
+		return 1
+	}
+	var err error
+	if timeout == 0 {
+		err = nodeManager.EnsureSync(context.Background())
+	} else {
+		ctx, cancel := context.WithTimeout(context.Background(), (time.Duration)(timeout)*time.Minute)
+		err = nodeManager.EnsureSync(ctx)
+		defer cancel()
+	}
+	if err != nil {
+		log.Printf("Sync error: %v", err)
+		exitCode = 1
+	}
+	var done <-chan struct{}
+	done, err = nodeManager.StopNode()
+	if err != nil {
+		log.Printf("Stop node err: %v", err)
+		exitCode = 1
+	}
+	<-done
```

Comment:
Not sure I was precise enough. If `err` occurs during `StopNode()` call, we should return from the `if` statement. This still can blow up if `err` occurred as `done` may be `nil`.

```go
var done <-chan struct{}
done, err = nodeManager.StopNode()
if err != nil {
  log.Printf("Stop node err: %v", err)
  return 1
}
<-done
return
```

Reply:
Done, sorry.